server/routes/visit/processVisitRequestController.ts

```diff
@@ -0,0 +1,87 @@
+import { RequestHandler } from 'express'
+import { VisitRequestsService, VisitService } from '../../services'
+import { MoJAlert } from '../../@types/bapv'
+import { convertToTitleCase } from '../../utils/utils'
+import { VisitBookingDetails, VisitRequestResponse } from '../../data/orchestrationApiTypes'
+
+type RequestAction = 'approve' | 'reject'
+
+export default class ProcessVisitRequestController {
+  public constructor(
+    private readonly visitRequestsService: VisitRequestsService,
+    private readonly visitService: VisitService,
+  ) {}
+
+  public processRequest(action: RequestAction): RequestHandler {
```

Comment:
Logic for approve and reject the same so handling together with `action` parameter determining which API call and message variant.